### DISCUSSION TITLE
Calling package function (pl/sql)

### DISCUSSION THREAD

**Discussion by Webblurt** [sijms/go-ora#571] [Q&A]
I tried to call package function trough this query:

```
query := "BEGIN :1 := MYDB.PACKAGE.MYFUNC(:2); END;"
_, err := tx.Exec(query, go_ora.Out{Dest: &variable_out}, variable_in)
```

but my variable_in passed as null to the func, I have IF ELSE construction in the MYFUNC and it returns 

`RAISE_APPLICATION_ERROR(-20001, 'Undefined nom name');` 

if variable_in IS NULL
so I logged variable_in and it is not nil it have value

also I tried to exec my query this way:

`_, err := tx.Exec(query, go_ora.Out{Dest: &variable_out, Size: 32}, variable_in)`

but it returns another error

`call register type before use user defined type (UDT)`

I do not use any UDT in MYFUNC so i don't know why it's happening 
what am i doing wrong?

**Comment by sijms** [comment]
what are the data types of the variables

**Reply by Webblurt** [reply]
:1 = CHAR(32), :2 = VARCHAR2 (in db)
variable_out and variable_in both string
